cmd/lncli/main.go

```diff
@@ -88,3 +139,17 @@ func main() {
 		fatal(err)
 	}
 }
+
+// cleanAndExpandPath expands environment variables and leading ~ in the
+// passed path, cleans the result, and returns it.
+func cleanAndExpandPath(path string) string {
+	// Expand initial ~ to OS specific home directory.
+	if strings.HasPrefix(path, "~") {
+		homeDir := filepath.Dir(lndHomeDir)
+		path = strings.Replace(path, "~", homeDir, 1)
```

Comment:
An edge case: if `path` contains `~` as part of file name or dir name (ex. `~/hello~world/ssl.cert`) this will replace more than once, leading to a bug